src/Esprima/Scanner.cs

```diff
@@ -517,7 +517,6 @@ internal ArrayList<Comment> ScanCommentsInternal()
     public ReadOnlySpan<Comment> ScanComments()
     {
         var comments = ScanCommentsInternal();
-        comments.TrimExcess();
```

Comment:
IMO we'd better trim the collected `comments` one way or the other because it's exposed in the public API.

However, if you don't see the point in reducing memory footprint and don't want to remove excess, then `ScanCommentsInternal` is unnecessary.